### PR TITLE
[fix-5453](pydevlake): Make ScopeConfig optional

### LINKED ISSUE #5453: [Bug][Pydevlake] Python should allow no ScopeConfig configuration
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

Current PyDevlake enforces that a ScopeConfig needs to be set per Scope. This should not be mandatory per @Startrekzky.
I have fixed the Go side of the code in #5449.

### What do you expect to happen

Python will throw an error like this if ScopeConfig is not set:

image

This code should be made flexible to allow this possiblity.

### How to reproduce

See above

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CamilleTeruel**
### Summary
Make ScopeConfig optional in pydevlake

### Does this close any open issues?
Closes #5453

### CODE REVIEW

File: backend/server/services/remote/plugin/plugin_impl.go [hunk 1]
```diff
@@ -176,8 +177,10 @@ func (p *remotePluginImpl) getScopeAndConfig(db dal.Dal, connectionId uint64, sc
 		if err != nil {
 			return nil, nil, err
 		}
+		return wrappedScope.Unwrap(), wrappedScopeConfig.Unwrap(), nil
+	} else {
+		return wrappedScope.Unwrap(), nil, nil
```
Comment (keon94) [inline backend/server/services/remote/plugin/plugin_impl.go:182]: Was ```wrappedScopeConfig.Unwrap()``` not just resolving to nil here?

Comment (keon94) [reply backend/server/services/remote/plugin/plugin_impl.go:182]: If you want to break the code up like this, I suggest initializing `wrappedScopeConfig` inside the if statement to keep its scope limited to there.

Comment (CamilleTeruel) [reply backend/server/services/remote/plugin/plugin_impl.go:182]: > Was `wrappedScopeConfig.Unwrap()` not just resolving to nil here?

No this resolves to a zeroed scope config.

**Comment by CamilleTeruel** [comment]
One thing I considered was to force scope configs to only have fields with defaults. When a config is not set we set a scope config with only defaults. Like that plugin devs don't have to check whether its nil everytime.

**Comment by klesh** [comment]
> One thing I considered was to force scope configs to only have fields with defaults. When a config is not set we set a scope config with only defaults. Like that plugin devs don't have to check whether its nil everytime.

That is how Golang plugin handles it, it is much easy, especially when the scope config is missing and we have to check the DataEntities. Note that we assuming users wants all DataEntities for missing scope config or `entities` is empty or null

**Comment by CamilleTeruel** [comment]
> That is how Golang plugin handles it, it is much easy, especially when the scope config is missing and we have to check the DataEntities. Note that we assuming users wants all DataEntities for missing scope config or `entities` is empty or null

Alright, I changed the PR to do the same thing.

**Review by keon94 — APPROVED** [review]

**Pull request merged**